internal/docker/context/context.go

```diff
@@ -0,0 +1,109 @@
+package context
+
+// The code in this file has been extracted from https://github.com/docker/cli,
+// more especifically from https://github.com/docker/cli/blob/master/cli/context/store/metadatastore.go
+// with the goal of not consuming the CLI package and all its dependencies.
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/testcontainers/testcontainers-go/internal/docker/config"
+	"github.com/testcontainers/testcontainers-go/internal/docker/context/internal"
+)
+
+const (
+	// DefaultContextName is the name reserved for the default context (config & env based)
+	DefaultContextName = "default"
+
+	// EnvOverrideContext is the name of the environment variable that can be
+	// used to override the context to use. If set, it overrides the context
+	// that's set in the CLI's configuration file, but takes no effect if the
+	// "DOCKER_HOST" env-var is set (which takes precedence.
+	EnvOverrideContext = "DOCKER_CONTEXT"
+
+	// EnvOverrideHost is the name of the environment variable that can be used
+	// to override the default host to connect to (DefaultDockerHost).
+	//
+	// This env-var is read by FromEnv and WithHostFromEnv and when set to a
+	// non-empty value, takes precedence over the default host (which is platform
+	// specific), or any host already set.
+	EnvOverrideHost = "DOCKER_HOST"
+
+	// contextsDir is the name of the directory containing the contexts
+	contextsDir = "contexts"
+
+	// metadataDir is the name of the directory containing the metadata
+	metadataDir = "meta"
+)
+
+// ErrDockerHostNotSet is the error returned when the Docker host is not set in the Docker context
+var ErrDockerHostNotSet = internal.ErrDockerHostNotSet
+
+// getContextFromEnv returns the context name from the environment variables.
+func getContextFromEnv() string {
+	if os.Getenv(EnvOverrideHost) != "" {
+		return DefaultContextName
+	}
+
+	if ctxName := os.Getenv(EnvOverrideContext); ctxName != "" {
+		return ctxName
+	}
+
+	return ""
+}
+
+// Current returns the current context name, based on
+// environment variables and the cli configuration file. It does not
+// validate if the given context exists or if it's valid.
+//
+// If the current context is not found, it returns the default context name.
+func Current() (string, error) {
```

Comment:
question: is current the best name? It seems odd as all functions are typically point in time, possibly Name or something similar.

Reply:
I've seen current and active as vamos in the Docker docs, although current seems part of the response when setting a context:

```shell 
docker context use docker-test
docker-test
Current context is now "docker-test"
```

Reply:
Also, the docker-config file contains this entry: `"currentContext": "value",`

Please check out with `cat ~/.docker/config.json`

Reply:
It's not wrong, and as you said matches the use elsewhere, I just found it odd / superfluous.